resources/qml/PrintSetupSelector/Custom/CustomPrintSetup.qml

```diff
@@ -75,62 +75,94 @@ Item
 
             baselineOffset: null // If we don't do this, there is a binding loop. WHich is a bit weird, since we override the contentItem anyway...
 
-            contentItem: Label
+
+            contentItem: RowLayout
             {
-                id: textLabel
-                text: intentSelection.text
-                font: UM.Theme.getFont("default")
-                color: UM.Theme.getColor("text")
+                spacing: 0
                 anchors.left: parent.left
+                anchors.right: customisedSettings.left
                 anchors.leftMargin: UM.Theme.getSize("default_margin").width
-                anchors.verticalCenter: intentSelection.verticalCenter
-                height: contentHeight
-                verticalAlignment: Text.AlignVCenter
-                renderType: Text.NativeRendering
-            }
 
-            background: Rectangle
-            {
-                id: backgroundItem
-                border.color: intentSelection.hovered ? UM.Theme.getColor("setting_control_border_highlight") : UM.Theme.getColor("setting_control_border")
-                border.width: UM.Theme.getSize("default_lining").width
-                radius: UM.Theme.getSize("default_radius").width
-                color: UM.Theme.getColor("main_background")
-            }
 
-            function generateActiveQualityText()
-            {
-                var resultMap = Cura.MachineManager.activeQualityDisplayNameMap
-                var resultMain = resultMap["main"]
-                var resultSuffix = resultMap["suffix"]
-                var result = ""
 
-                if (Cura.MachineManager.isActiveQualityExperimental)
+                Label
                 {
-                    resultSuffix += " (Experimental)"
+                    id: textLabel
+                    text: qualityName()
+                    font: UM.Theme.getFont("default")
+                    color: UM.Theme.getColor("text")
+                    anchors.verticalCenter: intentSelection.verticalCenter
+                    Layout.margins: 0
+                    Layout.maximumWidth: parent.width * 0.7
+                    height: contentHeight
+                    verticalAlignment: Text.AlignVCenter
+                    renderType: Text.NativeRendering
+                    elide: Text.ElideRight
+
+                    function qualityName() {
+                        var resultMap = Cura.MachineManager.activeQualityDisplayNameMap
+                        return resultMap["main"]
+                    }
```

Comment:
You could also write this as `Cura.MachineManager.activeQualityDisplayNameMap["main"]`, then it can be back to be a binding.

Also, if the function is this small, and only used in one place, you can also put the brackets directly where the value was. So instead of
~~~
thing: blah()
function blah() { ... bloeh... }
~~~
you just get:
~~~
thing: { ... bloeh ... }
~~~